### PR TITLE
enable softcap and gemma2

### PR DESCRIPTION
**Pull request by hliuca**
Gemma2 model needs softcap feature from flash attention.

### CODE REVIEW

File: vllm/attention/backends/rocm_flash_attn.py [hunk 1]
```diff
@@ -218,12 +218,6 @@ def decode_metadata(self) -> Optional["ROCmFlashAttentionMetadata"]:
             max_encoder_seq_len=self.max_encoder_seq_len,
             cross_slot_mapping=self.cross_slot_mapping,
             cross_block_tables=self.cross_block_tables)
-        # Batch may be composed of prefill|decodes, adjust query start indices
```
Comment (gshtras) [inline vllm/attention/backends/rocm_flash_attn.py:221]: Why is this section being removed?

File: vllm/attention/backends/rocm_flash_attn.py [hunk 2]
```diff
@@ -459,10 +453,12 @@ def __init__(
...
             raise ValueError(
                 "ROCmFlashAttention does not support blocksparse attention.")
-        if logits_soft_cap is not None:
-            raise ValueError(
-                "ROCmFlashAttention does not support attention logits soft "
-                "capping.")
+
+        if logits_soft_cap is None:
+            # In flash-attn, setting logits_soft_cap as 0 means no soft cap.
+            logits_soft_cap = 0
```
Comment (gshtras) [inline vllm/attention/backends/rocm_flash_attn.py:459]: This effectively enables logits_soft_cap for any model, unconditionally.

File: vllm/attention/backends/rocm_flash_attn.py [hunk 3]
```diff
@@ -511,6 +515,11 @@ def __init__(
                     self.use_naive_attn = True
 
             if self.use_naive_attn:
+                if logits_soft_cap is not None:
```
Comment (gshtras) [inline vllm/attention/backends/rocm_flash_attn.py:518]: You make sure it's not None in the constructor. So naive flash attention can never be used now

**Review by gshtras — CHANGES_REQUESTED** [review]
The linter warnings are due to attn_func always being called with the softcap parameter, but not all implementations support it.
Per the other comments please see to only using this parameter on the models that require it

File: vllm/attention/backends/rocm_flash_attn.py (hunk 1 above)
Comment (hliuca) [reply vllm/attention/backends/rocm_flash_attn.py]: sorry... this is accident... it has been restored.

File: vllm/attention/backends/rocm_flash_attn.py (hunk 2 above)
Comment (hliuca) [reply vllm/attention/backends/rocm_flash_attn.py:465]: In flash attention, it is 0 by default, I think. https://github.com/ROCm/flash-attention/blob/main/flash_attn/flash_attn_interface.py#L1334

File: vllm/attention/backends/rocm_flash_attn.py (hunk 3 above)
Comment (hliuca) [reply vllm/attention/backends/rocm_flash_attn.py:525]: fixed. thanks.

**Review by gshtras — APPROVED** [review]

**Pull request merged**